### PR TITLE
Match get_attr when compare node

### PR DESCRIPTION
**Pull request by cccclai**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91657

The pattern can't be matched if one attribute is `_param_constant1` and the other is `_param_constant0`

Large graph:
```
        # call_function  addmm_default      aten.addmm.default  (_param_constant1, ph_0, _tensor_constant0)  {}
```

Pattern graph
```
        # call_function  addmm_default      aten.addmm.default  (_param_constant0, ph_0, _tensor_constant0)  {}
```

Differential Revision: [D42316574](https://our.internmc.facebook.com/intern/diff/D42316574/)

### CODE REVIEW

File: torch/fx/passes/utils/matcher_utils.py [hunk 1]
```diff
@@ -99,6 +100,10 @@ def _nodes_are_equal(self, pn: Node, gn: Node) -> bool:
         if pn.op == gn.op:
             if pn.op == "placeholder" or pn.op == "output":
                 return True
+            elif pn.op == "get_attr":
+                pn_value = getattr(pn.graph.owning_module, pn.target)
+                gn_value = getattr(gn.graph.owning_module, gn.target)
+                return isinstance(pn_value, torch.Tensor) and isinstance (gn_value, torch.Tensor)
```
Comment (cccclai) [inline torch/fx/passes/utils/matcher_utils.py:106]: I feel like we don't necessarily need to compare tensor data itself, torch.linear(10, 14) and torch.linear(12, 14) will have different weight and they should be matched, thought? 

Also regarding
> legal object
anything else I need to check here?

File: torch/fx/passes/utils/matcher_utils.py [hunk 2]
```diff
@@ -91,6 +92,18 @@ def __init__(self, pattern: Graph,
             # and should be matched against as an anchor
             self.pattern_anchors = [n for n in output_node.all_input_nodes if len(n.users) == 1]
 
+    def _match_attributes(self, pn: Node, gn: Node) -> bool:
+        # Attributes matching is compilcated. Right now we only support matching constant tensor
+        pn_value = getattr(pn.graph.owning_module, pn.target)
+        gn_value = getattr(gn.graph.owning_module, gn.target)
+        if type(pn_value) != type(gn_value):
+            return False
+        if isinstance(pn_value, torch.Tensor):
```
Comment (SherlockNoMad) [inline torch/fx/passes/utils/matcher_utils.py:101]: Please comment on we don't require exact match on tensor values.

File: torch/fx/passes/utils/matcher_utils.py [hunk 3]
```diff
@@ -91,6 +92,18 @@ def __init__(self, pattern: Graph,
...
+            return isinstance (gn_value, torch.Tensor)
+        else:
```
Comment (SherlockNoMad) [inline torch/fx/passes/utils/matcher_utils.py:103]: We should be able handle python literals, int/float/bool ect...

**Comment by SherlockNoMad** [comment]
Can we add some test?

**Review by SherlockNoMad — APPROVED** [review]
Per comment and lint passing.

File: torch/fx/passes/utils/matcher_utils.py (hunk 3 above)
Comment (cccclai) [reply torch/fx/passes/utils/matcher_utils.py:103]: Make sense. For python literals, i'm thinking they should be pass if the type is the same maybe?

**Pull request closed without merging**